language/variables_spec.rb

```diff
@@ -797,17 +797,6 @@ module VariableSpecs
 end
 
 describe 'Allowed characters' do
-  # new feature in 2.6 -- https://bugs.ruby-lang.org/issues/13770
-  it 'does not allow non-ASCII upcased characters at the beginning' do
-    -> do
-      eval <<-CODE
-        def test
-          ἍBB = 1
-        end
-      CODE
-    end.should raise_error(SyntaxError, /dynamic constant assignment/)
-  end
```

Comment:
The title is completely incorrect - the mentioned fix was about allowing constants to start with a non-ASCII character.

Reply:
I think it's related to below, and that the Ruby implementation correctly notices this is upcased and so not a valid local variable name.
Could you add it back, after the spec below and change the description to something like:
> parses a non-ASCII upcased character as a constant identifier

We could also add a spec to actually set the constant in a valid context, e.g., on some module, that seems useful.

Reply:
> We could also add a spec to actually set the constant in a valid context, e.g., on some module, that seems useful.

Do you mean something like this one?

https://github.com/ruby/spec/blob/12c704ac53d4200610d48c50e0e5d1c4dd40aa82/language/constants_spec.rb#L714-L719

Reply:
> Could you add it back, after the spec below and change the description to something like:

Done

Reply:
> Do you mean something like this one?

Yes, but that ones uses `const_set` which might behave differently. I think it'd be useful to have the same but with a literal constant assignment within a named module.